src/process_watch.js

```diff
@@ -0,0 +1,369 @@
+  //currentTime = new Date().getTime();
+  var pidStored = function (stored) {
+    if (stored == undefined) {
+      return false;
+    }
+    return processObject.pid == stored.pid;
+  };
+  let found = 0;
+  pidusage(processObject.pid, function (err, stats) {
+    // all processes always get updated with this information
+    processObject.elapsed = toMinutes(stats.elapsed);
+    processObject.memory = formatBytes(stats.memory);
+
+  })
+  if (processWarnList.some(pidStored)) {
+    processObject.lastDetectTime = currentTime;
+    //if exists in storage
+    // update; cpu, memory, lastDetectTime, 
+
+    //element.lastDetectTime <= (currentTime - reportInterval)
+    index = _.findIndex(processWarnList, function (element) {
+      return element.pid == processObject.pid
+    })
+    /* 
+     * found time, Show how long a ps has been on list
+     * age should be difference between found and current time 
+     */
+    processObject.firstDetectTime = processWarnList[index].firstDetectTime;
+    // reportTime, Show how long since last report
+    processObject.reportTime = processWarnList[index].reportTime;
+    // If a new process, that is repeat offender, allow reporting it
+    // Also a little over 15 min old 
+    processObject.report = ((processObject.reportTime == undefined) && (
+      processObject.firstDetectTime <= (currentTime - 15 * 60 * 1000)));
+    //overwrite previous
+    processWarnList.splice(index, 1, processObject);
+  } else {
+    processObject.lastDetectTime = currentTime;
+    processObject.firstDetectTime = currentTime;
+    processObject.report = false;
+    processWarnList.push(processObject);
+  }
+}
+
+module.exports = {
+  /**
+   * init
+   * @param {ps} current-processes npm module
+   * @param {pid} pidusage npm module
+   * @param {steam} output stream for reports. Usually Slack. STDout for testing
+   * bring in the process watching modules
+   * AND the stream for output to Slack
+   */
+  init: function (ps, pid, stream) {
+    psLookup = ps;
+    pidusage = pid;
+    currStream = stream;
+  },
+
+  /**
+   * checkProcess
+   * @param {currentTime} used track in lists when 'new' and when 'updated'
+   * On first new run we check cpu type
+   * 
+   * new cpuAddLevel processes are added to Watch list
+   * If a  PID 'match' is in Watch already, process is added to Warn list
+   * 
+   * If tracked process is above cpuAddLevel, it's 'last detected' timestamp is updated
+   * If a process is below cpuAddLevel, it can eventually age out of the list
+   * 
+   * 'cleanList' is then run with 'processes,' a current snapshot of all processes.
+   */
+  checkProcess: function checkProcess(currentTime = new Date().getTime()) {
+    // stops function from crashing server if init hasn't been run
+    if (!psLookup) {
+      console.log("process_watch not initialized");
+      return null;
+    }
+    cpuTypeChecked = false;
+    psLookup.get(function (err, processes) {
```

Comment:
do you check for err?

What would happen if this returns an error?

Reply:
Now checking for err, it'll be logged and will returned